binary_search_tree/tree.py

```diff
@@ -14,42 +17,131 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def add_helper(self, current_node, key, value):
+        if current_node == None:
+            return TreeNode(key, value)
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(current_node.right, key, value)
+        return current_node
+
+
+    # Time Complexity:O(log n) 
+    # Space Complexity:O(1)
     def add(self, key, value = None):
-        pass
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
+
+
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n) 
+    # Space Complexity: O(1)
     def find(self, key):
-        pass
+        if self.root == None:
+            return None
+        
+        current = self.root
+        while current != None:
+            if current.key == key:
+                return current.value
+            elif current.key < key:
+                current = current.right
+            else:
+                current = current.left
+        return None
+    def inorder_helper_f(self,current_node, in_order_list):
```

Comment:
👍 Good helper function

for spacing

```suggestion

    def inorder_helper_f(self,current_node, in_order_list):
```